pkg/agent/core/revsource/revsource.go

```diff
@@ -0,0 +1,131 @@
+package revsource
+
+import (
+	"errors"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	vmv1 "github.com/neondatabase/autoscaling/neonvm/apis/neonvm/v1"
+)
+
+const (
+	Upscale vmv1.Flag = 1 << iota
+	Downscale
+	Immediate
+)
+
+// AllFlags and AllFlagNames must have the same order, so the metrics work correctly.
+var AllFlags = []vmv1.Flag{Upscale, Downscale, Immediate}
+var AllFlagNames = []string{"upscale", "downscale", "immediate"}
```

Comment:
Same comments here as before

Reply:
Responding:

> I also don't quite get why the clock needs to care about scaling flags -- feels like it's breaking separation of concerns. But presumably I'm missing something?

> edit: OTOH, an alternate approach would be making LogicClock generic over the flag type, and then providing the flags when it's instantiated in agent/core or wherever.

Yeah, currently, the flag type is defined together with other NeonVM types (because it is used in the CRD), but the actual flags and their meaning defined in revsource. This gives a little bit of separation of concern, and kinda works like generics. Bringing full-fledged generics here might be a bad idea, because:
1. CRD generation from generic types will probably break
2. I try to avoid generics in the business logic code

Reply:
I'll look into how other metrics in the agent expose this info and will see if I can do something similar.

Reply:
Previous discussion: https://github.com/neondatabase/autoscaling/pull/983#discussion_r1669835735

Reply:
Reused directionLabel in be700fff